airflow/sensors/external_task.py

```diff
@@ -199,18 +199,17 @@ def poke(self, context, session=None):
         dttm_filter = self._get_dttm_filter(context)
         serialized_dttm_filter = ",".join(dt.isoformat() for dt in dttm_filter)
 
-        if self.external_task_ids:
+        if self.external_task_group_id:
             self.log.info(
-                "Poking for tasks %s in dag %s on %s ... ",
-                self.external_task_ids,
+                "Poking for task_group '%s' in dag '%s' on %s ... ",
+                self.external_task_group_id,
                 self.external_dag_id,
                 serialized_dttm_filter,
             )
-
-        if self.external_task_group_id:
+        else:
             self.log.info(
-                "Poking for task_group '%s' in dag '%s' on %s ... ",
-                self.external_task_group_id,
+                "Poking for tasks %s in dag %s on %s ... ",
```

Comment:
I thin, we should change the log and  it should be "Poking for DAG"  if task_id is None according to documentation of the Sensor. We should likely also fali the sensor when both  task_id and group_id are specified.

Reply:
Thanks for the review, will work on the changes.